agent/agent.go

```diff
@@ -482,6 +474,14 @@ func (a *Agent) connectionManager(ctx context.Context, cancel context.CancelFunc
 func (a *Agent) receiveLoop(ctx context.Context, cancel context.CancelFunc, conn transport.Transport) {
 	defer cancel()
 	for {
+		select {
+		case <-ctx.Done():
+			if err := conn.Close(); err != nil {
+				logger.WithError(err).Error("error closing websocket connection")
+			}
+			return
+		default:
+		}
```

Comment:
This can be written more simply as:
```
if err := ctx.Err(); err != nil {
    if err := conn.Close(); err != nil {
        logger.WithError(err).Error("error closing websocket connection")
    }
    return
}
```

Reply:
Done!